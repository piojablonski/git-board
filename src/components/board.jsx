import React from 'react'
import { connect } from 'react-redux'

export const BoardComponent = () => <div>board</div>

const mapStateToProps = (state) => ({})
const mapDispatchToProps = (dispatch) => ({})

export const Board = connect(mapStateToProps, mapDispatchToProps)(BoardComponent)