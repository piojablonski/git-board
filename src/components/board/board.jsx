import React from 'react'
import { connect } from 'react-redux'

export const BoardComponent = ({ data }) => (
  [
    <div key='1'>board</div>,
    <table key='2'>
      <thead>
        <tr>
          <th>title</th>
          <th>body</th>
          <th>assignee</th>
          <th></th>
        </tr>
      </thead>
      <tbody>
        {data && data.map(issue => (
          <tr key={issue.id}>
            <td>{issue.title}</td>
            <td>{issue.body}</td>
            <td>{issue.assignee && issue.assignee.login}</td>
            <td><a href={issue.html_url} target='_blank'>open</a></td>
          </tr>
        ))
        }
      </tbody>
    </table>
  ]
)

const mapStateToProps = (state) => ({ data: state.issues.issues })
const mapDispatchToProps = (dispatch) => ({})

export const Board = connect(mapStateToProps, mapDispatchToProps)(BoardComponent)
