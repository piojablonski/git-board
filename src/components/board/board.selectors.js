export const optionsSelector = (issuesState) => {
  if (!issuesState) {
    return undefined
  }
  if (!issuesState.options.atom) {
    return undefined
  }
  const { static: { state, sort, direction, assignees: staticAssignees }, atom: { labels, milestones, assignees } } = issuesState.options
  const res = {
    state,
    sort,
    direction,
    labels,
    milestones,
    creators: assignees,
    assignees: [].concat(staticAssignees, assignees)
  }
  return res
}
