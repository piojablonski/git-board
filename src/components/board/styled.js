import styled from 'styled-components'

const sidebarWidth = '320px'
const headerHeight = '60px'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: stretch;
  height: 100%;
`
export const DataWrapper = styled.div`
  margin-top: ${headerHeight};
  padding: 20px;
  margin-left: ${sidebarWidth};
`

export const TableWrapper = styled.div`
  color: green;
  tbody {
    color: tomato;
  }
`

export const Sidebar = styled.div`
  display: flex;
  flex-direction: column;
  width: ${sidebarWidth};
  height: 100vh;
  background-color: white;
  border-right: 1px solid grey;
  position: fixed;
`

export const FilterWrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding: 1rem;
  > div, input {
    margin-bottom: 1rem;
  }
  > label {
    margin-bottom: .15rem;
    margin-left: .15rem;
    &:after {
      content: ':';
    }
  }

  > .ant-calendar-picker {
    width: 100% !important;
    .ant-calendar-picker-clear,
    .ant-calendar-picker-icon {
      top: 35%;
    }
  }
`

export const TagListWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  > * {
    margin-bottom: 0.25rem
  }
  `

export const ColumnInfoWrapper = styled.div`
  display: flex;
  flex-direction: column;
  > header {
    margin-top:0.15rem;
    font-size: 0.75em;
    font-weight: normal;
    &:after {
      content: ':';
    }
  }
  * {
    margin-bottom: 0.25rem;
    font-weight: bold;
  }
`

export const User = styled.div`
  display: flex;
  align-items: center;
  line-height: 24px;
  > span:first-child {
    margin-right: 0.25rem
  }
`

export const HeaderWrapper = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-end;
  position: fixed;
  width: calc(100% - ${sidebarWidth});
  margin-left: ${sidebarWidth};
  background-color: gray;
  height: 50px;
  > * {
    margin-right: 0.5rem;
  }

`
