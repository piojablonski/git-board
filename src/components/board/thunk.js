import { issuesActions } from '../../reducers/issues.reducer'
import { redirect, NOT_FOUND } from 'redux-first-router'
import { get } from '../../api'

const mapDataToOption = (value, name) => res => res.data.map(l => ({ value: l[value], name: l[name] })) || []
const sanitizeSelectedFilters = selectedFilters => {
  const res = Object.keys(selectedFilters).reduce((acc, key) => {
    const value = selectedFilters[key]
    if (Array.isArray(value)) {
      acc[key] = value.join()
    } else {
      acc[key] = value
    }
    return acc
  }, {})
  return res
}

export const boardThunk = async (dispatch, getState) => {
  const { issues: { options, repo, selectedFilters } } = getState()
  const shouldLoadOptions = !options[repo]
  const getData = get(repo)

  if (shouldLoadOptions) {
    Promise.all([
      getData('labels', { params: { per_page: 100 } }).then(mapDataToOption('id', 'name')),
      getData('milestones', { params: { per_page: 100 } }).then(mapDataToOption('id', 'title')),
      getData('assignees', { params: { per_page: 100 } }).then(mapDataToOption('login', 'login'))
    ]).then(([labels, milestones, assignees]) => {
      const options = ({
        labels,
        milestones,
        assignees
      })
      dispatch(issuesActions.receivedOptions({ repo, options }))
    })
  }

  getData('issues', {
    params: sanitizeSelectedFilters(selectedFilters)
  }).then(response => {
    if (response.data) {
      dispatch(issuesActions.receivedData(response.data))
    } else {
      throw response.error
    }
  }).catch(() => {
    // dispatch(redirect({ type: NOT_FOUND }))
  })
}
