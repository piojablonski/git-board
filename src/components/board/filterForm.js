import React from 'react'
import { Select, Button } from 'antd'
import { FilterWrapper } from './styled';
const { Option } = Select

export const FilterForm = ({ options, selectedFilters, filterChangedHandler, filtersApplyHandler }) => {
  return (
    <FilterWrapper>
      <label>state</label>
      <Select
        defaultValue={selectedFilters.state}
        onChange={value => filterChangedHandler(value, 'state')} >
        {options.state.map(s => <Option key={s.value} value={s.value}>{s.name}</Option>)}
      </Select>
      <label>assignee</label>
      <Select placeholder='choose...'
        showSearch
        defaultValue={selectedFilters.assignee}
        onChange={value => filterChangedHandler(value, 'assignee')} >
        {options.assignees.map(s => <Option key={s.value} value={s.value}>{s.name}</Option>)}
      </Select>
      <label>creator</label>
      <Select placeholder='choose...'
        showSearch
        allowClear
        defaultValue={selectedFilters.creator}
        onChange={value => filterChangedHandler(value, 'creator')} >
        {options.creators.map(s => <Option key={s.value} value={s.value}>{s.name}</Option>)}
      </Select>
      <label>label</label>
      <Select placeholder='choose...'
        showSearch
        allowClear
        mode='tags'
        defaultValue={selectedFilters.labels}
        onChange={value => filterChangedHandler(value, 'labels')} >
        {options.labels.map(s => <Option key={s.value} value={s.name}>{s.name}</Option>)}
      </Select>
      <Button onClick={filtersApplyHandler}>Search</Button>
    </FilterWrapper>
  )
}
