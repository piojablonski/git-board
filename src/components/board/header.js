import React from 'react'
import { connect } from 'react-redux'
import { redirect } from 'redux-first-router'
import { issuesActions } from '../../reducers/issues.reducer'
import { HeaderWrapper } from './styled'
import { Button, Select, Icon } from 'antd'
import { navigate } from '../../utils/utils'
const { Option } = Select

export const HeaderComponent = ({ pagination, pageSizeOptions, filterChangeAndRedirect, selectedFilters }) => {
  const executeChange = (key) => (value) => {
    filterChangeAndRedirect(selectedFilters, value, 'per_page', { page: 1 })
  }
  const executeChangePage = ({ target: { value } }) => {
    filterChangeAndRedirect(selectedFilters, value, 'page')
  }
  return (
    <HeaderWrapper>
      <Button type="primary" disabled={!pagination.hasPrevious} value={pagination.previousPage} onClick={executeChangePage}>
        <Icon type="left" />Previous
      </Button>
      <span>{`${pagination.page} / ${pagination.lastPage}`}</span>
      <Button type="primary" disabled={!pagination.hasNext} value={pagination.nextPage} onClick={executeChangePage}>
        Next<Icon type="right" />
      </Button>
      <Select
        value={pagination.perPage}
        onChange={executeChange('per_page')} >
        {pageSizeOptions.map(p => <Option key={p} value={p}>{p}</Option>)}
      </Select>
    </HeaderWrapper>
  )
}

const mapStateToProps = (state) => ({
  pageSizeOptions: state.issues.options.static.perPage,
  pagination: paginationInfoSelector(state.issues),
  selectedFilters: state.issues.selectedFilters

})
const mapDispatchToProps = (dispatch) => ({
  filterChangeAndRedirect: (query, value, filterKey, extraValues = {}) => {
    dispatch(navigate('ISSUES', { ...query, ...extraValues, [filterKey]: value }))
  }
})

const paginationInfoSelector = issuesState => {
  const { lastPage, selectedFilters: { page, per_page } } = issuesState
  const nLastPage = Number(lastPage)
  const nPage = Number(page)
  const res = ({
    hasPrevious: nPage > 1,
    previousPage: String(nPage - 1),
    currentPage: page,
    lastPage,
    page,
    hasNext: nPage < nLastPage,
    nextPage: String(nPage + 1),
    perPage: per_page
  })
  return res
}

export const Header = connect(mapStateToProps, mapDispatchToProps)(HeaderComponent)
