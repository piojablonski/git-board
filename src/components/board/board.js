import { issuesActions } from '../../reducers/issues.reducer'
import React from 'react'
import { connect } from 'react-redux'
import { optionsSelector } from './board.selectors'
import { redirect } from 'redux-first-router'
import { Sidebar, Wrapper, DataWrapper, TagListWrapper, User, UserListWrapper } from './styled'
import { FilterForm } from './filterForm'
import { Table, Tag, Avatar } from 'antd'

const UserItem = ({ user }) => <User><Avatar size='small' src={user.avatar_url} /><span>{user.login}</span></User>

export const BoardComponent = ({ data, options, selectedFilters, filterChangedHandler, filtersApplyHandler }) => {
  if (!data || !options) {
    return null
  }

  const columns = [
    {
      title: 'Title',
      key: 'title',
      render: record => <a href={record.html_url} target='_blank'>{record.title}</a>
    },
    { title: 'State', width: 80, dataIndex: 'state', key: 'state' },
    {
      title: 'Assignees',
      width: 120,
      key: 'assignees',
      render: record => (<UserListWrapper>
        <span>author:</span>
        <UserItem user={record.user} />
        <span>assignees:</span>
        {record.assignees.map(a => <UserItem key={a.id} user={a} />)}
      </UserListWrapper >)
    },
    {
      title: 'Labels',
      key: 'Labels',
      render: record => (
        <TagListWrapper>
          {record.labels.map(l => <Tag key={l.id} color={`#${l.color}`}>{l.name}</Tag>)}
        </TagListWrapper>)
    }
  ]

  return <Wrapper>
    <Sidebar>
      <h1>Repo</h1>
      <FilterForm
        options={options}
        selectedFilters={selectedFilters}
        filterChangedHandler={filterChangedHandler}
        filtersApplyHandler={() => filtersApplyHandler(selectedFilters)} />
    </Sidebar>
    <DataWrapper>
      <Table
        columns={columns}
        dataSource={data}
        pagination={false}
        rowKey='id'
        expandedRowRender={record => <p>{record.body}</p>}
      />
    </DataWrapper>
  </Wrapper>
}

const mapStateToProps = (state) => ({
  data: state.issues.issues,
  options: optionsSelector(state.issues),
  selectedFilters: state.issues.selectedFilters
})
const mapDispatchToProps = (dispatch) => ({
  filterChangedHandler: (value, filterKey) => {
    dispatch(issuesActions.filterChanged({ value, filterKey }))
  },
  filtersApplyHandler: query => {
    dispatch(redirect({ type: 'ISSUES', query }))
  }
})

export const Board = connect(mapStateToProps, mapDispatchToProps)(BoardComponent)
