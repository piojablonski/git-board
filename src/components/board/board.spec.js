import React from 'react'
import { BoardComponent } from './board'
import { mount } from 'enzyme'
import { matchSnapshot } from '../../utils/testUtils'
import mockGitResponse from '../../mockData/git-issues-response.json'

describe('BoardComponent snapshots', () => {
  test('no data', () => {
    const comp = mount(<BoardComponent />)
    matchSnapshot(comp)
  })

  test('with data', () => {
    const comp = mount(<BoardComponent data={mockGitResponse} />)
    matchSnapshot(comp)
  })
})
