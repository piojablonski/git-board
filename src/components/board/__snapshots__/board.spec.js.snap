// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardComponent snapshots no data 1`] = `<Component />`;

exports[`BoardComponent snapshots with data 1`] = `
<Component
  data={
    Array [
      Object {
        "assignee": Object {
          "avatar_url": "https://avatars3.githubusercontent.com/u/118951?v=4",
          "events_url": "https://api.github.com/users/damieng/events{/privacy}",
          "followers_url": "https://api.github.com/users/damieng/followers",
          "following_url": "https://api.github.com/users/damieng/following{/other_user}",
          "gists_url": "https://api.github.com/users/damieng/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/damieng",
          "id": 118951,
          "login": "damieng",
          "organizations_url": "https://api.github.com/users/damieng/orgs",
          "received_events_url": "https://api.github.com/users/damieng/received_events",
          "repos_url": "https://api.github.com/users/damieng/repos",
          "site_admin": true,
          "starred_url": "https://api.github.com/users/damieng/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/damieng/subscriptions",
          "type": "User",
          "url": "https://api.github.com/users/damieng",
        },
        "assignees": Array [
          Object {
            "avatar_url": "https://avatars3.githubusercontent.com/u/118951?v=4",
            "events_url": "https://api.github.com/users/damieng/events{/privacy}",
            "followers_url": "https://api.github.com/users/damieng/followers",
            "following_url": "https://api.github.com/users/damieng/following{/other_user}",
            "gists_url": "https://api.github.com/users/damieng/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/damieng",
            "id": 118951,
            "login": "damieng",
            "organizations_url": "https://api.github.com/users/damieng/orgs",
            "received_events_url": "https://api.github.com/users/damieng/received_events",
            "repos_url": "https://api.github.com/users/damieng/repos",
            "site_admin": true,
            "starred_url": "https://api.github.com/users/damieng/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/damieng/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/damieng",
          },
        ],
        "author_association": "NONE",
        "body": "
    
    ### Prerequisites
    
    * [X ] Put an X between the brackets on this line.
    
    ### Description
    
    The fonts are looking strange,
    I tried to reinstall atom three times (including the .atom directory) manually,
    Tried to run on safe mode, development mode, and disabled gpu.
    
    ### Steps to Reproduce
    
    1. Install Northen Dark Syntax And UI Themes.
    2. Use Them.
    3. Restart atom and see.
    
    **Expected behavior:** [What you expect to happen]
    
    **Actual behavior:** [What actually happens]
    
    **Reproduces how often:** [What percentage of the time does it reproduce?]
    
    ### Versions
    
    OS: Void Linux x86_64
    Model: H55M-D2H
    Kernel: 4.12.13_1
    Shell: zsh 5.4.2
    Resolution: 1920x1080
    CPU: Intel i3 540 (4) @ 3.059GHz
    GPU: NVIDIA GeForce GTX 650
    Memory: 7915MiB
    
    apm --version:
    apm  1.18.4
    npm  3.10.10
    node 6.9.5 x64
    python 2.7.14
    git 2.14.1
    
    atom --version:
    Atom    : 1.20.1
    Electron: 1.6.9
    Chrome  : 56.0.2924.87
    Node    : 7.4.0
    
    ### Additional Information
    ![2017-09-24-204828_1920x1080_scrot](https://user-images.githubusercontent.com/25435369/30784005-78c321f4-a155-11e7-8af5-9cea9abab6ec.png)
    
    ",
        "closed_at": null,
        "comments": 41,
        "comments_url": "https://api.github.com/repos/atom/atom/issues/15737/comments",
        "created_at": "2017-09-24T15:23:40Z",
        "events_url": "https://api.github.com/repos/atom/atom/issues/15737/events",
        "html_url": "https://github.com/atom/atom/issues/15737",
        "id": 260092687,
        "labels": Array [
          Object {
            "color": "A5ECFA",
            "default": false,
            "id": 258255692,
            "name": "electron",
            "url": "https://api.github.com/repos/atom/atom/labels/electron",
          },
          Object {
            "color": "c7def8",
            "default": false,
            "id": 97577987,
            "name": "linux",
            "url": "https://api.github.com/repos/atom/atom/labels/linux",
          },
          Object {
            "color": "eb6420",
            "default": false,
            "id": 343564310,
            "name": "typography",
            "url": "https://api.github.com/repos/atom/atom/labels/typography",
          },
        ],
        "labels_url": "https://api.github.com/repos/atom/atom/issues/15737/labels{/name}",
        "locked": false,
        "milestone": null,
        "number": 15737,
        "repository_url": "https://api.github.com/repos/atom/atom",
        "state": "open",
        "title": "Font Rendering issue with freetype 2.8.1",
        "updated_at": "2017-11-17T00:10:25Z",
        "url": "https://api.github.com/repos/atom/atom/issues/15737",
        "user": Object {
          "avatar_url": "https://avatars3.githubusercontent.com/u/25435369?v=4",
          "events_url": "https://api.github.com/users/3xpl0it2c/events{/privacy}",
          "followers_url": "https://api.github.com/users/3xpl0it2c/followers",
          "following_url": "https://api.github.com/users/3xpl0it2c/following{/other_user}",
          "gists_url": "https://api.github.com/users/3xpl0it2c/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/3xpl0it2c",
          "id": 25435369,
          "login": "3xpl0it2c",
          "organizations_url": "https://api.github.com/users/3xpl0it2c/orgs",
          "received_events_url": "https://api.github.com/users/3xpl0it2c/received_events",
          "repos_url": "https://api.github.com/users/3xpl0it2c/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/3xpl0it2c/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/3xpl0it2c/subscriptions",
          "type": "User",
          "url": "https://api.github.com/users/3xpl0it2c",
        },
      },
      Object {
        "assignee": Object {
          "avatar_url": "https://avatars3.githubusercontent.com/u/1058982?v=4",
          "events_url": "https://api.github.com/users/Ben3eeE/events{/privacy}",
          "followers_url": "https://api.github.com/users/Ben3eeE/followers",
          "following_url": "https://api.github.com/users/Ben3eeE/following{/other_user}",
          "gists_url": "https://api.github.com/users/Ben3eeE/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/Ben3eeE",
          "id": 1058982,
          "login": "Ben3eeE",
          "organizations_url": "https://api.github.com/users/Ben3eeE/orgs",
          "received_events_url": "https://api.github.com/users/Ben3eeE/received_events",
          "repos_url": "https://api.github.com/users/Ben3eeE/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/Ben3eeE/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/Ben3eeE/subscriptions",
          "type": "User",
          "url": "https://api.github.com/users/Ben3eeE",
        },
        "assignees": Array [
          Object {
            "avatar_url": "https://avatars3.githubusercontent.com/u/1058982?v=4",
            "events_url": "https://api.github.com/users/Ben3eeE/events{/privacy}",
            "followers_url": "https://api.github.com/users/Ben3eeE/followers",
            "following_url": "https://api.github.com/users/Ben3eeE/following{/other_user}",
            "gists_url": "https://api.github.com/users/Ben3eeE/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/Ben3eeE",
            "id": 1058982,
            "login": "Ben3eeE",
            "organizations_url": "https://api.github.com/users/Ben3eeE/orgs",
            "received_events_url": "https://api.github.com/users/Ben3eeE/received_events",
            "repos_url": "https://api.github.com/users/Ben3eeE/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/Ben3eeE/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/Ben3eeE/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/Ben3eeE",
          },
        ],
        "author_association": "NONE",
        "body": "[Enter steps to reproduce:]
    
    1. Open any file associated with atom editor from Gnome
    2. Type anything in the editor - crashes
    3. Open atom from terminal, no crash.
    
    **Atom**: 1.20.0 x64
    **Electron**: 1.6.9
    **OS**: linux 4.12.12-1-ARCH
    **Thrown From**: Atom Core
    
    
    ### Stack Trace
    
    Uncaught Error: Could not connect to X display.
    
    \`\`\`
    At /usr/share/atom/resources/app/node_modules/keyboard-layout/lib/keyboard-layout.js:16
    
    Error: Could not connect to X display.
        at get_manager (/usr/share/atom/resources/app/node_modules/keyboard-layout/lib/keyboard-layout.js:16:37)
        at getCurrentKeyboardLayout (/usr/share/atom/resources/app/node_modules/keyboard-layout/lib/keyboard-layout.js:32:17)
        at Object.getCurrentKeymap (/usr/share/atom/resources/app/node_modules/keyboard-layout/lib/keyboard-layout.js:22:31)
        at exports.keystrokeForKeyboardEvent (/usr/share/atom/resources/app/node_modules/atom-keymap/lib/helpers.js:244:66)
        at KeymapManager.module.exports.KeymapManager.keystrokeForKeyboardEvent (/usr/share/atom/resources/app/node_modules/atom-keymap/lib/keymap-manager.js:467:20)
        at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/usr/share/atom/resources/app/node_modules/atom-keymap/lib/keymap-manager.js:347:30)
        at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/usr/share/atom/resources/app/src/window-event-handler.js:100:42)
        at HTMLDocument.<anonymous> (/usr/share/atom/resources/app/src/window-event-handler.js:3:65)
    \`\`\`
    
    ### Commands
    
    \`\`\`
      2x -0:06.6.0 core:select-right (input.hidden-input)
         -0:06.1.0 intentions:highlight (input.hidden-input)
    \`\`\`
    
    ### Non-Core Packages
    
    \`\`\`
    atom-beautify 0.30.5 
    autocomplete-clang 0.11.3 
    busy-signal 1.4.3 
    git-plus 7.9.3 
    intentions 1.1.5 
    language-cmake 0.3.0 
    linter 2.2.0 
    linter-clang 4.1.2 
    linter-ui-default 1.6.8 
    platformio-ide-terminal 2.6.0 
    \`\`\`
    
    ",
        "closed_at": null,
        "comments": 3,
        "comments_url": "https://api.github.com/repos/atom/atom/issues/15647/comments",
        "created_at": "2017-09-13T23:21:25Z",
        "events_url": "https://api.github.com/repos/atom/atom/issues/15647/events",
        "html_url": "https://github.com/atom/atom/issues/15647",
        "id": 257549662,
        "labels": Array [
          Object {
            "color": "c7def8",
            "default": false,
            "id": 97577987,
            "name": "linux",
            "url": "https://api.github.com/repos/atom/atom/labels/linux",
          },
          Object {
            "color": "009800",
            "default": false,
            "id": 171329633,
            "name": "needs-reproduction",
            "url": "https://api.github.com/repos/atom/atom/labels/needs-reproduction",
          },
        ],
        "labels_url": "https://api.github.com/repos/atom/atom/issues/15647/labels{/name}",
        "locked": false,
        "milestone": null,
        "number": 15647,
        "repository_url": "https://api.github.com/repos/atom/atom",
        "state": "open",
        "title": "Uncaught Error: Could not connect to X display",
        "updated_at": "2017-10-04T19:00:04Z",
        "url": "https://api.github.com/repos/atom/atom/issues/15647",
        "user": Object {
          "avatar_url": "https://avatars0.githubusercontent.com/u/7622286?v=4",
          "events_url": "https://api.github.com/users/elimpnick/events{/privacy}",
          "followers_url": "https://api.github.com/users/elimpnick/followers",
          "following_url": "https://api.github.com/users/elimpnick/following{/other_user}",
          "gists_url": "https://api.github.com/users/elimpnick/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/elimpnick",
          "id": 7622286,
          "login": "elimpnick",
          "organizations_url": "https://api.github.com/users/elimpnick/orgs",
          "received_events_url": "https://api.github.com/users/elimpnick/received_events",
          "repos_url": "https://api.github.com/users/elimpnick/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/elimpnick/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/elimpnick/subscriptions",
          "type": "User",
          "url": "https://api.github.com/users/elimpnick",
        },
      },
      Object {
        "assignee": Object {
          "avatar_url": "https://avatars3.githubusercontent.com/u/118951?v=4",
          "events_url": "https://api.github.com/users/damieng/events{/privacy}",
          "followers_url": "https://api.github.com/users/damieng/followers",
          "following_url": "https://api.github.com/users/damieng/following{/other_user}",
          "gists_url": "https://api.github.com/users/damieng/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/damieng",
          "id": 118951,
          "login": "damieng",
          "organizations_url": "https://api.github.com/users/damieng/orgs",
          "received_events_url": "https://api.github.com/users/damieng/received_events",
          "repos_url": "https://api.github.com/users/damieng/repos",
          "site_admin": true,
          "starred_url": "https://api.github.com/users/damieng/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/damieng/subscriptions",
          "type": "User",
          "url": "https://api.github.com/users/damieng",
        },
        "assignees": Array [
          Object {
            "avatar_url": "https://avatars3.githubusercontent.com/u/118951?v=4",
            "events_url": "https://api.github.com/users/damieng/events{/privacy}",
            "followers_url": "https://api.github.com/users/damieng/followers",
            "following_url": "https://api.github.com/users/damieng/following{/other_user}",
            "gists_url": "https://api.github.com/users/damieng/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/damieng",
            "id": 118951,
            "login": "damieng",
            "organizations_url": "https://api.github.com/users/damieng/orgs",
            "received_events_url": "https://api.github.com/users/damieng/received_events",
            "repos_url": "https://api.github.com/users/damieng/repos",
            "site_admin": true,
            "starred_url": "https://api.github.com/users/damieng/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/damieng/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/damieng",
          },
        ],
        "author_association": "NONE",
        "body": "### Prerequisites
    
    * [ ] Put an X between the brackets on this line if you have done all of the following:
        * Reproduced the problem in Safe Mode: http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
        * Followed all applicable steps in the debugging guide: http://flight-manual.atom.io/hacking-atom/sections/debugging/
        * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
        * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
        * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages
    
    ### Description
    
    Sometimes when updating packages, Atom loses most of its settings. E.g. right now, I updated Hyperclick, and after the restart I was greeted with the welcome message, my font was reset to the default settings, my settings for linter-pycodestyle and linter-pylint were lost, and I assume all the others as well, but I didn't bother with the rest much, so no impact there in my case.
    
    This is difficult to reproduce; I can update packages dozens of times, but every once in a while, it loses all my settings. It's not a disk full issue, as I have 56GB available.
    
    The data loss appears to be limited to the things in ~/.atom/config.cson - the other .cson files and styles.less appear to be unaffected.
    
    ### Steps to Reproduce
    
    1. Update packages
    2. Restart
    
    **Expected behavior:** Atom restarts with upgraded packages, and settings intact.
    
    **Actual behavior:** Atom restarts with upgraded packages, and lost settings.
    
    **Reproduces how often:** Less than 10%
    
    ### Versions
    Atom    : 1.18.0
    Electron: 1.3.15
    Chrome  : 52.0.2743.82
    Node    : 6.5.0
    apm  1.18.1
    npm  3.10.10
    node 6.9.5 x64
    python 2.7.6
    git 1.9.1
    
    OS: Linux Mint 17.3 Rosa
    Kernel: 3.19.0-32-generic #37~14.04.1-Ubuntu SMP Thu Oct 22 09:41:40 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
    
    ### Additional Information
    
    Any additional information, configuration or data that might be necessary to reproduce the issue.
    ",
        "closed_at": null,
        "comments": 38,
        "comments_url": "https://api.github.com/repos/atom/atom/issues/14909/comments",
        "created_at": "2017-06-28T12:48:55Z",
        "events_url": "https://api.github.com/repos/atom/atom/issues/14909/events",
        "html_url": "https://github.com/atom/atom/issues/14909",
        "id": 239152969,
        "labels": Array [
          Object {
            "color": "f7c6c7",
            "default": true,
            "id": 33602150,
            "name": "bug",
            "url": "https://api.github.com/repos/atom/atom/labels/bug",
          },
          Object {
            "color": "e11d21",
            "default": false,
            "id": 372218233,
            "name": "data-loss",
            "url": "https://api.github.com/repos/atom/atom/labels/data-loss",
          },
          Object {
            "color": "009800",
            "default": false,
            "id": 98444358,
            "name": "more-information-needed",
            "url": "https://api.github.com/repos/atom/atom/labels/more-information-needed",
          },
          Object {
            "color": "009800",
            "default": false,
            "id": 171329633,
            "name": "needs-reproduction",
            "url": "https://api.github.com/repos/atom/atom/labels/needs-reproduction",
          },
          Object {
            "color": "00bb00",
            "default": false,
            "id": 505038679,
            "name": "triaged",
            "url": "https://api.github.com/repos/atom/atom/labels/triaged",
          },
        ],
        "labels_url": "https://api.github.com/repos/atom/atom/issues/14909/labels{/name}",
        "locked": false,
        "milestone": null,
        "number": 14909,
        "repository_url": "https://api.github.com/repos/atom/atom",
        "state": "open",
        "title": "Settings lost",
        "updated_at": "2017-10-23T17:14:15Z",
        "url": "https://api.github.com/repos/atom/atom/issues/14909",
        "user": Object {
          "avatar_url": "https://avatars1.githubusercontent.com/u/570247?v=4",
          "events_url": "https://api.github.com/users/jrial/events{/privacy}",
          "followers_url": "https://api.github.com/users/jrial/followers",
          "following_url": "https://api.github.com/users/jrial/following{/other_user}",
          "gists_url": "https://api.github.com/users/jrial/gists{/gist_id}",
          "gravatar_id": "",
          "html_url": "https://github.com/jrial",
          "id": 570247,
          "login": "jrial",
          "organizations_url": "https://api.github.com/users/jrial/orgs",
          "received_events_url": "https://api.github.com/users/jrial/received_events",
          "repos_url": "https://api.github.com/users/jrial/repos",
          "site_admin": false,
          "starred_url": "https://api.github.com/users/jrial/starred{/owner}{/repo}",
          "subscriptions_url": "https://api.github.com/users/jrial/subscriptions",
          "type": "User",
          "url": "https://api.github.com/users/jrial",
        },
      },
    ]
  }
/>
`;
